subprojects/core/src/integTest/groovy/org/gradle/api/tasks/UntrackedPropertiesIntegrationTest.groovy

```diff
@@ -490,6 +485,69 @@ class UntrackedPropertiesIntegrationTest extends AbstractIntegrationSpec impleme
         LocalState | 'ConfigurableFileCollection'
     }
 
+    def "invalidates the VFS for untracked output directories"() {
+        buildFile("""
+            abstract class ProducerWithUntrackedOutput extends DefaultTask {
+                @Untracked
+                @OutputDirectory
+                abstract DirectoryProperty getOutputDir()
+
+                @TaskAction
+                void createOutput() {
+                    outputDir.file("untracked.txt").get().asFile.text = "untracked"
+                }
+            }
```

Comment:
Can you reuse `generateProducerTask()` for this?

Reply:
Actually, I think you should be able to reuse the methods for each of these tasks.

Reply:
Perhaps by passing a default `name` parameter for the tasks to the methods.

Reply:
I don't think I can re-use the methods. The producer tasks generate unreadable files no matter if the output is tracked or not. That is not what I want, since I want the consumer to have a tracked input.